### ISSUE TITLE
Type error when trying to conditionally show fields using resolveFields

### ISSUE THREAD

**Issue by the-code-raccoon** [measuredco/puck#530]
## Background

> Puck version: 0.15.0

## Current Behaviour

I have a component where I would like to show `field A` conditionally based on the value of `field B`. I created a type for the component as follows:
```
type MyComponentProps {
    fieldA?: string;
    fieldB: boolean;
}
```
My component looks like this:
```
const MyComponent = (): ComponentConfig<MyComponentProps> => {
     ... ,
     resolveFields: (data, { fields }) => {
          if (data.props.fieldB) {
               return { ... fields, fieldA: undefined } 
          }
          return fields
     }
}
```
But then I receive the following type error on `resolveFields`:
`Types of property 'fieldA' are incompatible. Type 'undefined' is not assignable to type 'Field<string | undefined>'`

Note `field A` does renders conditionally as desired even with the type error.

## Expected Behaviour
I would assume this method of showing fields conditionally would work considering I found this method for conditionally showing fields in the puck demo part of the repo.

**Comment by chrisvxd** [comment]
Good catch @the-code-raccoon! This is is an issue with how the types for component fields are defined. It's assumed that every prop is required as a field, regardless of whether or not the prop is optional. This was okay before we had conditional fields, because it wouldn't have made sense to include a prop without a field, but doesn't make sense with dynamic fields.

I'll mark this as a bug.

**Comment by nickfla1-mxm** [comment]
Hi @chrisvxd, is there a plan to fix this? If not, or not so soon, would you accept contributions?

**Comment by chrisvxd** [comment]
I would have thought this should have been resolved in 5a219ef, released in v0.16.0. I'm not sure why the bug is still occurring, but happy to accept contributions as I'm preoccupied resolving #598

**Comment by FedericoBonel** [comment]
I cannot reproduce this in the latest versions, so I will be closing this as complete.
